cloudinit/config/cc_raspberry_pi.py

```diff
@@ -0,0 +1,220 @@
+RPI_BASE_KEY = "rpi"
+RPI_INTERFACES_KEY = "interfaces"
+ENABLE_RPI_CONNECT_KEY = "enable_rpi_connect"
+SUPPORTED_INTERFACES = {
+    "spi": "do_spi",
+    "i2c": "do_i2c",
+    "serial": "do_serial",
+    "onewire": "do_onewire",
+    "remote_gpio": "do_rgpio",
+}
+RASPI_CONFIG_SERIAL_CONS_FN = "do_serial_cons"
+RASPI_CONFIG_SERIAL_HW_FN = "do_serial_hw"
+
+meta: MetaSchema = {
+    "id": "cc_raspberry_pi",
+    "distros": ["raspberry-pi-os"],
+    "frequency": PER_INSTANCE,
+    "activate_by_schema_keys": [RPI_BASE_KEY],
+}
+
+
+def configure_rpi_connect(enable: bool) -> None:
+    LOG.debug("Configuring rpi-connect: %s", enable)
+
+    num = 0 if enable else 1
+
+    try:
+        subp.subp(["/usr/bin/raspi-config", "do_rpi_connect", str(num)])
+    except subp.ProcessExecutionError as e:
+        LOG.error("Failed to configure rpi-connect: %s", e)
+
+
+def is_pifive() -> bool:
+    try:
+        subp.subp(["/usr/bin/raspi-config", "nonint", "is_pifive"])
+        return True
+    except subp.ProcessExecutionError:
+        return False
+
+
+def configure_serial_interface(
+    cfg: Union[dict, bool], instCfg: Config, cloud: Cloud
+) -> None:
+    def get_bool_field(cfg_dict: dict, name: str, default=False):
+        val = cfg_dict.get(name, default)
+        if not isinstance(val, bool):
+            LOG.warning(
+                "Invalid value for %s.serial.%s: %s",
+                RPI_INTERFACES_KEY,
+                name,
+                val,
+            )
+            return default
+        return val
+
+    enable_console = False
+    enable_hw = False
+
+    if isinstance(cfg, dict):
+        enable_console = get_bool_field(cfg, "console")
+        enable_hw = get_bool_field(cfg, "hardware")
+
+    elif isinstance(cfg, bool):
+        # default to enabling console as if < pi5
+        # this will also enable the hardware
+        enable_console = cfg
+
+    if not is_pifive() and enable_console:
+        # only pi5 has 2 usable UARTs
+        # on other models, enabling the console
+        # will also block the other UART
+        enable_hw = True
+
+    try:
+        subp.subp(
+            [
+                "/usr/bin/raspi-config",
+                "nonint",
+                RASPI_CONFIG_SERIAL_CONS_FN,
+                str(0 if enable_console else 1),
+            ]
+        )
+
+        try:
+            subp.subp(
+                [
+                    "/usr/bin/raspi-config",
+                    "nonint",
+                    RASPI_CONFIG_SERIAL_HW_FN,
+                    str(0 if enable_hw else 1),
+                ]
+            )
+        except subp.ProcessExecutionError as e:
+            LOG.error("Failed to configure serial hardware: %s", e)
+
+        # Reboot to apply changes
+        cmd = cloud.distro.shutdown_command(
+            mode="reboot",
+            delay="now",
+            message="Rebooting to apply serial console changes",
+        )
+        _ = subprocess.Popen(
```

Comment:
Sorry, looking at this again, can we use `subp.subp()` here like above? It provides consistent logging and error handling for spawning subprocesses. Also no need to save the response to `_`. Just call it without assigning it.